EIPS/eip-7804.md

```diff
@@ -0,0 +1,220 @@
+| `WITHDRAWAL_CREDENTIALS_UPDATE_REQUEST_QUEUE_TAIL_STORAGE_SLOT` | 3 | Pointer to the tail of the withdrawal credential update request message queue|
+| `WITHDRAWAL_CREDENTIALS_UPDATE_REQUEST_QUEUE_STORAGE_OFFSET` | 4 | The start memory slot of the in-state withdrawal credential update request message queue|
+| `MAX_WITHDRAWAL_CREDENTIALS_UPDATE_REQUESTS_PER_BLOCK` | 4 | Maximum number of withdrawal credential update requests that can be dequeued into a block |
+| `TARGET_WITHDRAWAL_CREDENTIALS_UPDATE_REQUESTS_PER_BLOCK` | 1 | |
+| `MIN_WITHDRAWAL_CREDENTIALS_UPDATE_REQUEST_FEE` | 1 | |
+| `WITHDRAWAL_CREDENTIALS_UPDATE_REQUEST_FEE_UPDATE_FRACTION` | 17 | |
+| `EXCESS_INHIBITOR` | `2**256-1` | Excess value used to compute the fee before the first system call |
+
+### Execution Layer
+
+* **`FORK_BLOCK`** -- the first block in a blockchain with the `timestamp` greater or equal to `FORK_TIMESTAMP`.
+
+#### Withdrawal Credentials Update request operation
+
+The new withdrawal credential update request operation is an [EIP-7685](./eip-7685.md) request
+with type `0x03` and consists of the following fields:
+
+1. `pubkey`: `Bytes48`
+2. `old_address`: `Bytes20`
+3. `new_address`: `Bytes20`
+
+The [EIP-7685](./eip-7685.md) encoding of a withdrawal credential update request is computed as follows.
+
+```python
+request_type = WITHDRAWAL_CREDENTIALS_UPDATE_REQUEST_TYPE
+request_data = read_withdrawal_credential_update_requests()
+```
+
+#### Withdrawal Credentials Update Request Contract
+
+The contract is similar to other execution requests contracts for Deposits, Withdrawals and Consolidation.
+The contract has three different code paths, which can be summarized at a high level as follows:
+
+1. Add request - requires a `68` byte input, the validator's public key concatenated with the new address for withdrawal credentials.
+2. Excess requests getter - if the input length is zero, return the current excess requests count.
+3. System process - if called by system address, pop off the withdrawal credential update requests for the current block from the queue.
+
+<!-- TODO add sudo code / bytecode / deployment / etc_ (very similar to existing request contracts) -->
+
+#### Block processing
+
+At the end of processing any execution block where `block.timestamp >= FORK_TIMESTAMP` (i.e. after processing all transactions and after performing the block body requests validations) client software **MUST** include a call the contract as `SYSTEM_ADDRESS` and empty input data to trigger the system subroutine execute. The resopnse should be treated as a new request type (0x03) according to [EIP-7685](./eip-7685.md).
+
+#### Block Validation
+
+EL must check that the commitment hash in the execution block header matches the hash of the list of execution requests the CL sends when
+validating the execution block (including any requests of the the new defined type 0x03).
+
+### Consensus Layer
+
+Full specification <!-- TODO add link to consensus-spec PR -->
```

Comment:
the specification here also looks half baked so can you add another `<!-- TODO complete the specification -->` so we can merge this for draft

Reply:
Of course! Done! :)